
---
title: "Data analysis: Nature vs Nurture in the rhizosphere"
author: "Alexa Byers"
date: "19/05/2022 (last edited)"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#load libraries
```{r}
library(vegan)
library(picante)
library(stats)
library(pairwiseAdonis)
library(phyloseq)
library(ggplot2)
library(dplyr)
library(stats)
library(microbiome)
library(tidyverse)
library(metagenomeSeq)
library(ape)
library(psych)
theme_set(theme_bw())
```
## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.
When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

#1. Set up phyloseq objects using outputs of dada2 processing 
#files were cleaned post dada2 to remove singletons, unidentified phyla, and Chloroplasts/Cyanobacteria
#load ITS data
```{r}
ASV.ITS <- read.csv("ITS_dada2/ASV_counts_cleaned.csv")
Taxa.ITS <- read.csv("ITS_dada2/Taxonomy_cleaned.csv")
Sample.ITS <- read.csv("ITS_dada2/ITS_samples.csv")
ITS.tree <- read.tree("ITS_fasttreeRooted/tree.nwk")
#format dataframes
row.names(ASV.ITS) = ASV.ITS$ASVs
ASV.ITS = ASV.ITS[,-1:-2]
row.names(Taxa.ITS) = Taxa.ITS$ASVs
Taxa.ITS = Taxa.ITS[-1:-2]
row.names(Sample.ITS) = Sample.ITS$SampleID
```
# create ITS phyloseq object
```{r}
ASVS.ITS = otu_table(as.matrix(ASV.ITS), taxa_are_rows = TRUE)
TAXA.ITS = tax_table(as.matrix(Taxa.ITS))
SAMPLES.ITS = sample_data(Sample.ITS)
ITS.tree = phy_tree(ITS.tree)
physeq_ITS = phyloseq(ASVS.ITS, TAXA.ITS, SAMPLES.ITS, ITS.tree)
```
#load 16S data
```{r read in and prepare 16S data}
ASV.16S <- read.csv("16S_dada2/ASV_counts_cleaned.csv")
Taxa.16S <- read.csv("16S_dada2/Taxonomy_cleaned.csv")
Sample.16S <- read.csv("16S_dada2/16S_samples_post_filtering.csv")
x16S.tree <- read.tree("16s_fasttreeRooted/tree.nwk")

#format datafrmaes
ASV.16S = ASV.16S[,-1]
row.names(ASV.16S) = ASV.16S$ASV_ID
ASV.16S = ASV.16S[,-1]
row.names(Taxa.16S) = Taxa.16S$ASV_ID
Taxa.16S = Taxa.16S[-1:-2]
row.names(Sample.16S) = Sample.16S$SampleID
```
#create 16S phyloseq object
```{r}
ASVS.16S = otu_table(as.matrix(ASV.16S), taxa_are_rows = TRUE)
TAXA.16S = tax_table(as.matrix(Taxa.16S))
SAMPLES.16S = sample_data(Sample.16S)
x16S.tree = phy_tree(x16S.tree)
physeq.16S = phyloseq(ASVS.16S, TAXA.16S, SAMPLES.16S, x16S.tree)
```
#rarefy ITS ASV counts
```{r}
#plot rarefaction curve
ASVs.dat.ITS = as(otu_table(physeq_ITS), "matrix")
if(taxa_are_rows(physeq_ITS)){ASVs.dat.ITS <- t(ASVs.dat.ITS)}
ASVdf.ITS = as.data.frame(ASVs.dat.ITS)
ASVs.sample.ITS <- data.frame(sample_data(physeq_ITS))

ASVdf.ITS <- cbind(rownames(ASVdf.ITS), data.frame(ASVdf.ITS, row.names = NULL))
colnames(ASVdf.ITS)[1] <- "SampleID"
row.names(ASVdf.ITS) = ASVdf.ITS$SampleID

png(filename="ITS_rarecurve.png", width=2500, height =2000, res=200)
rarecurve(ASVdf.ITS[-1], step = 20, xlab = "Number of Sequences", ylab = "ASVs")

##remove samples with low read counts (less than 9000)
Samples_toRemove_ITS <- c("kabir_ITS2rcbc567", "kabir_ITS2rcbc556", "kabir_ITS2rcbc459", "kabir_ITS2rcbc403", "kabir_ITS2rcbc555", "kabir_ITS2rcbc464")
physeq2_ITS <- subset_samples(physeq_ITS, !(SampleID %in% Samples_toRemove_ITS))
#remove ITS samples with no plant data
physeq3_ITS <- subset_samples(physeq2_ITS, PlantID !="NPD")
#remove ITS taxa which may now have 0 counts
physeq4_ITS <- prune_taxa(taxa_sums(physeq3_ITS) > 1 , physeq3_ITS)
any(taxa_sums(physeq4_ITS)== 0)

#rarefy ITS data to minimum sampling depth (5712 ASV counts)
physeq.rarefied_ITS = rarefy_even_depth(physeq4_ITS, rngseed=1, sample.size=1*min(sample_sums(physeq4_ITS)), replace=F) 
```
#ITS alpha diversity
```{r}
asv_its <- as.data.frame(otu_table(physeq.rarefied_ITS))
tree_its <- phy_tree(physeq.rarefied_ITS)
pd.diversity_ITS <- pd(t(asv_its), tree_its, include.root = TRUE) #calculate phylogenetic diversity
pd.diversity_ITS <- cbind(pd.diversity_ITS, sample_data(physeq.rarefied_ITS))
```
#summarise alpha diversity results for each plant metadata factor
```{r}
pd.diversity_ITS1 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$PlantID, mat = TRUE, digits=3)
pd.diversity_ITS2 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Genus, mat = TRUE, digits=3)
pd.diversity_ITS3 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Family, mat = TRUE, digits=3)
pd.diversity_ITS4 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Order, mat = TRUE, digits=3)
pd.diversity_ITS5 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Class, mat = TRUE, digits=3)
pd.diversity_ITS6 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Life.span, mat = TRUE, digits=3)
pd.diversity_ITS7 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Nitrogen.fixing, mat = TRUE, digits=3)
pd.diversity_ITS8 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Primary.mycorrhizal.association, mat = TRUE, digits=3)
pd.diversity_ITS9 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Provenance, mat = TRUE, digits=3)
pd.diversity_ITS10 <- describe.by(pd.diversity_ITS$PD, pd.diversity_ITS$Functional.group, mat = TRUE, digits=3)

#create single summary table
pd_diversity_summary <- rbind(pd.diversity_ITS1, pd.diversity_ITS2, pd.diversity_ITS3, pd.diversity_ITS4, pd.diversity_ITS5, pd.diversity_ITS6, pd.diversity_ITS7, pd.diversity_ITS8, pd.diversity_ITS9, pd.diversity_ITS10)

#species richness
SR.diversity_ITS1 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$PlantID, mat = TRUE, digits=3)
SR.diversity_ITS2 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Genus, mat = TRUE, digits=3)
SR.diversity_ITS3 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Family, mat = TRUE, digits=3)
SR.diversity_ITS4 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Order, mat = TRUE, digits=3)
SR.diversity_ITS5 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Class, mat = TRUE, digits=3)
SR.diversity_ITS6 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Life.span, mat = TRUE, digits=3)
SR.diversity_ITS7 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Nitrogen.fixing, mat = TRUE, digits=3)
SR.diversity_ITS8 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Primary.mycorrhizal.association, mat = TRUE, digits=3)
SR.diversity_ITS9 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Provenance, mat = TRUE, digits=3)
SR.diversity_ITS10 <- describe.by(pd.diversity_ITS$SR, pd.diversity_ITS$Functional.group, mat = TRUE, digits=3)

#create single summary table
SR_diversity_summary <- rbind(SR.diversity_ITS1, SR.diversity_ITS2, SR.diversity_ITS3, SR.diversity_ITS4, SR.diversity_ITS5, SR.diversity_ITS6, SR.diversity_ITS7, SR.diversity_ITS8, SR.diversity_ITS9, SR.diversity_ITS10)
```
#rarefy 16S ASV table
```{r}
#plot rarefaction curve
ASVs.dat.16S = as(otu_table(physeq.16S), "matrix")
if(taxa_are_rows(physeq.16S)){ASVs.dat.16S <- t(ASVs.dat.16S)}
ASVdf.16S = as.data.frame(ASVs.dat.16S)
ASVs.sample.16S <- data.frame(sample_data(physeq.16S))

ASVdf.16S <- cbind(rownames(ASVdf.16S), data.frame(ASVdf.16S, row.names = NULL))
colnames(ASVdf.16S)[1] <- "SampleID"
row.names(ASVdf.16S) = ASVdf.16S$SampleID

png(filename="x16s_rarecurve.png", width=2500, height =2000, res=200)
rarecurve(ASVdf.16S[-1], step = 20, xlab = "Number of Sequences", ylab = "ASVs")

#remove 16S samples with low read counts below 5000
Samples_toRemove_16S <- c("X128", "X530", "X183", "X646", "X740", "X117", "X642", "X580", "X23", "X513", "X625", "X156", "X669", "X626", "X714", "X586", "X87", "X641", "X226", "X549", "X120", "X453")
physeq2.16S <- subset_samples(physeq.16S, !(SampleID %in% Samples_toRemove_16S))
#remove 16S samples with no plant data
physeq3.16S <- subset_samples(physeq2.16S, PlantID !="NPD")
#remove 16S taxa which may now have < 1 count
physeq4.16S <- prune_taxa(taxa_sums(physeq3.16S) > 1 , physeq3.16S)
any(taxa_sums(physeq4.16S)== 0)

#rarefy 16S table to minimum sampling depth (5509 ASV counts)
physeq.rarefied.16S = rarefy_even_depth(physeq4.16S, rngseed=1, sample.size=1*min(sample_sums(physeq4.16S)), replace=F)
sample_sums(physeq.rarefied.16S) 
```
#alpha diversity on rarefied 16S data
```{r}
asv_16s <- as.data.frame(otu_table(physeq.rarefied.16S))
tree_16s <- phy_tree(physeq.rarefied.16S)
pd.diversity_16S <- pd(t(asv_16s), tree_16s, include.root = TRUE)
pd.diversity_16S <- cbind(pd.diversity_16S, sample_data(physeq.rarefied.16S))
```
#summarise alpha diversity results for each plant metadata factor
```{r}
pd.diversity_16S1 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$PlantID, mat = TRUE, digits=3)
pd.diversity_16S2 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Genus, mat = TRUE, digits=3)
pd.diversity_16S3 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Family, mat = TRUE, digits=3)
pd.diversity_16S4 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Order, mat = TRUE, digits=3)
pd.diversity_16S5 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Class, mat = TRUE, digits=3)
pd.diversity_16S6 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Life.span, mat = TRUE, digits=3)
pd.diversity_16S7 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Nitrogen.fixing, mat = TRUE, digits=3)
pd.diversity_16S8 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Primary.mycorrhizal.association, mat = TRUE, digits=3)
pd.diversity_16S9 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Provenance, mat = TRUE, digits=3)
pd.diversity_16S10 <- describe.by(pd.diversity_16S$PD, pd.diversity_16S$Functional.group, mat = TRUE, digits=3)

#create single summary table
pd_diversity_summary2 <- rbind(pd.diversity_16S1, pd.diversity_16S2, pd.diversity_16S3, pd.diversity_16S4, pd.diversity_16S5, pd.diversity_16S6, pd.diversity_16S7, pd.diversity_16S8, pd.diversity_16S9, pd.diversity_16S10)

#species richness
SR.diversity_16S1 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$PlantID, mat = TRUE, digits=3)
SR.diversity_16S2 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Genus, mat = TRUE, digits=3)
SR.diversity_16S3 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Family, mat = TRUE, digits=3)
SR.diversity_16S4 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Order, mat = TRUE, digits=3)
SR.diversity_16S5 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Class, mat = TRUE, digits=3)
SR.diversity_16S6 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Life.span, mat = TRUE, digits=3)
SR.diversity_16S7 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Nitrogen.fixing, mat = TRUE, digits=3)
SR.diversity_16S8 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Primary.mycorrhizal.association, mat = TRUE, digits=3)
SR.diversity_16S9 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Provenance, mat = TRUE, digits=3)
SR.diversity_16S10 <- describe.by(pd.diversity_16S$SR, pd.diversity_16S$Functional.group, mat = TRUE, digits=3)
#create single summary table
SR_diversity_summary2 <- rbind(SR.diversity_16S1, SR.diversity_16S2, SR.diversity_16S3, SR.diversity_16S4, SR.diversity_16S5, SR.diversity_16S6, SR.diversity_16S7, SR.diversity_16S8, SR.diversity_16S9, SR.diversity_16S10)
```
#kruskal wallis tests on 16S alpha diversity
```{r}
KW_16S_data <- data.frame(pd.diversity_16S[c(5:14)])
KW_16S_SR <- lapply(KW_16S_data, function(x) kruskal.test(pd.diversity_16S$SR ~ x))
KW_16S_PD <- lapply(KW_16S_data, function(x) kruskal.test(pd.diversity_16S$PD ~ x))
```
#pairwise wilcox tests
```{r}
PW_16S_SR <- lapply(KW_16S_data, function(x) pairwise.wilcox.test(pd.diversity_16S$SR, x, p.adjust.method = "bonferroni"))
PW_16S_PD <- lapply(KW_16S_data, function(x) pairwise.wilcox.test(pd.diversity_16S$PD, x, p.adjust.method = "bonferroni"))
```
###BETA DIVERSITY ANALYSIS
###ITS beta diversity analysis
#calculate ITS community compositon using weighted UniFrac distance matrix
```{r}
physeq.rarefied_ITS <- subset_samples(physeq.rarefied_ITS, SampleID !="kabir_ITS2rcbc295") #huge outliers
physeq.rarefied_ITS <- subset_samples(physeq.rarefied_ITS, SampleID !="kabir_ITS2rcbc532") 
unifrac.its <- phyloseq::distance(physeq.rarefied_ITS, "wunifrac")
ordination.its <- ordinate(physeq.rarefied_ITS, method = "NMDS", distance= unifrac.its)
sample.its <- data.frame(sample_data(physeq.rarefied_ITS))
```
#adonis and pairwise adonis testing
```{r}
Adonis_ITS_data <- sample.its[c(3:12)]
Adonis_unifrac_ITS <- lapply(Adonis_ITS_data, function(x) vegan::adonis2(unifrac.its ~ x))
Pairwise_adonis_ITS <- lapply(Adonis_ITS_data, function(x) pairwise.adonis(unifrac.its, x, p.adjust.m = "bonferroni", reduce = NULL, perm = 999))
```
#NMDS plots. Setting consistent layout style
```{r its nmds plots}
nmds_layout = 
    theme_bw() + 
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank())
```
```{r}
its_plot = plot_ordination(physeq.rarefied_ITS, ordination.its, type="samples", color="PlantID") +
    nmds_layout + 
    geom_point(size=4, alpha= 1)
```
#export plot
```{r}
png("FigureS3.png", width = 2500, height = 2000, res = 200)
its_plot
```
#mantel tests to correlate env variables with unifrac distances
#prep data
```{r}
sample.ITS2 <- data.frame(sapply(sample.its[,14:24], as.numeric))
rownames(sample.ITS2) <- rownames(sample.its)
its.vegdist <- lapply(sample.ITS2, vegdist, method="euclidean", na.rm = TRUE)
#mantel tests on weighted unifrac distances
Mantel.ITS <- lapply(its.vegdist, function(x) vegan::mantel(x, unifrac.its, method="spearman", permutations = 999, na.rm = TRUE))
```
##betadisper tests to analyse within group variance
```{r}
Betadisp_ITS_data <- sample.its[c(3:12)]
betadisp_its <- lapply(Betadisp_ITS_data, function(x) vegan::betadisper(unifrac.its, x, type = "median", bias.adjust = FALSE)) #calculating betadispersion
permu_betadisp_its <- lapply(betadisp_its, permutest, pairwise=TRUE) #permutation tests
#pairwise tests on betadispersion
Permu_betadisp_its.pairwise <- lapply(permu_betadisp_its, function (x) x$pairwise$permuted <- x$pairwise$permuted[!(x$pairwise$permuted > 0.05)]) #only keeping sig results for output
```
##betadispersion summary statistics
```{r}
BD_ITS.PlantID <- describe.by(betadisp_distances_ITS$PlantID, betadisp_distances_ITS$PlantID.1, mat = TRUE, digits=3)
BD_ITS.Genus <- describe.by(betadisp_distances_ITS$Genus, betadisp_distances_ITS$Genus.1, mat = TRUE, digits=3)
BD_ITS.Family <- describe.by(betadisp_distances_ITS$Family, betadisp_distances_ITS$Family.1, mat = TRUE, digits=3)
BD_ITS.Order <- describe.by(betadisp_distances_ITS$Order, betadisp_distances_ITS$Order.1, mat = TRUE, digits=3)
BD_ITS.Class <- describe.by(betadisp_distances_ITS$Class, betadisp_distances_ITS$Class.1, mat = TRUE, digits=3)
BD_ITS.Provenance <- describe.by(betadisp_distances_ITS$Provenance, betadisp_distances_ITS$Provenance.1, mat = TRUE, digits=3)
BD_ITS.PMA <- describe.by(betadisp_distances_ITS$Primary.mycorrhizal.association, betadisp_distances_ITS$Primary.mycorrhizal.association.1, mat = TRUE, digits=3)
BD_ITS.NF <- describe.by(betadisp_distances_ITS$Nitrogen.fixing, betadisp_distances_ITS$Nitrogen.fixing.1, mat = TRUE, digits=3)
BD_ITS.FG <- describe.by(betadisp_distances_ITS$Functional.group, betadisp_distances_ITS$Functional.group.1, mat = TRUE, digits=3)
BD_ITS.LS <- describe.by(betadisp_distances_ITS$Life.span, betadisp_distances_ITS$Life.span.1, mat = TRUE, digits=3)
#capture into one summary table
Betadispersion_its_summary <- rbind(BD_ITS.PlantID, BD_ITS.Genus, BD_ITS.Family, BD_ITS.Order, BD_ITS.Class, BD_ITS.Provenance, BD_ITS.PMA, BD_ITS.NF, BD_ITS.FG, BD_ITS.LS)
```
##16S beta diversity analysis
#calculating weighted UniFrac matrices
```{r}
unifrac.16S <- phyloseq::distance(physeq.rarefied.16S, "wunifrac")
ordination.16S <- ordinate(physeq.rarefied.16S, method = "NMDS", distance= unifrac.16S) #nmds ordination
sample.16S <- data.frame(sample_data(physeq.rarefied.16S))
```
#adonis and pairwise adonis testing
```{r}
Adonis_16S_data <- sample.16S[c(3:12)]
Adonis_tests_16S <- lapply(Adonis_16S_data, function(x) vegan::adonis2(unifrac.16S ~ x))
#pairwise adonis
Pairwise_adonis_16S <- lapply(Adonis_16S_data, function(x) pairwise.adonis(unifrac.16S, x, p.adjust.m = "bonferroni", reduce = NULL, perm = 999))
```
##16S nmds plots
```{r}
x16S_plot = plot_ordination(physeq.rarefied.16S, ordination.16S, type="samples", color="PlantID") +   
  nmds_layout + 
  geom_point(size=4, alpha= 1)
```
#export plot
```{r}
png("FigureS4.png", width = 2500, height = 2000, res = 200)
x16S_plot
```
#mantel tests to correlate env variables with weighted unifrac distances
```{r 16S mantel tests; data prep}
sample.16S2 <- data.frame(sapply(sample.16S[,14:24],as.numeric))
rownames(sample.16S2) = rownames(sample.16S2)
x16s.vegdist <- lapply(sample.16S2, vegdist, method="euclidean", na.rm = TRUE)
Mantel.16S <- lapply(x16s.vegdist, function(x) vegan::mantel(x, unifrac.16S, method="spearman", permutations = 999, na.rm = TRUE))
```
##16S betadisper tests to test within group variance
```{r}
Betadisp_16S_data <- sample.16S[c(3:12)]
Betadisp_16S_data$PlantID <- as.factor(Betadisp_16S_data$PlantID)
levels(Betadisp_16S_data$PlantID)[levels(Betadisp_16S_data$PlantID)=="F.novae-zelandiae "] <- "F.nz" #formatting issue needed corrected
betadisp_16S <- lapply(Betadisp_16S_data, function(x) vegan::betadisper(unifrac.16S, x, type = "median", bias.adjust = FALSE))
#permutation tests on betadispersion scores
permu_betadisp_16s <- lapply(betadisp_16S, permutest, pairwise=TRUE)
```
##betadispersion summary statistics
```{r}
BD_16S.PlantID <- describe.by(betadisp_distances_16S$PlantID, betadisp_distances_16S$PlantID.1, mat = TRUE, digits=3)
BD_16S.Genus <- describe.by(betadisp_distances_16S$Genus, betadisp_distances_16S$Genus.1, mat = TRUE, digits=3)
BD_16S.Family <- describe.by(betadisp_distances_16S$Family, betadisp_distances_16S$Family.1, mat = TRUE, digits=3)
BD_16S.Order <- describe.by(betadisp_distances_16S$Order, betadisp_distances_16S$Order.1, mat = TRUE, digits=3)
BD_16S.Class <- describe.by(betadisp_distances_16S$Class, betadisp_distances_16S$Class.1, mat = TRUE, digits=3)
BD_16S.Provenance <- describe.by(betadisp_distances_16S$Provenance, betadisp_distances_16S$Provenance.1, mat = TRUE, digits=3)
BD_16S.PMA <- describe.by(betadisp_distances_16S$Primary.mycorrhizal.association, betadisp_distances_16S$Primary.mycorrhizal.association.1, mat = TRUE, digits=3)
BD_16S.NF <- describe.by(betadisp_distances_16S$Nitrogen.fixing, betadisp_distances_16S$Nitrogen.fixing.1, mat = TRUE, digits=3)
BD_16S.FG <- describe.by(betadisp_distances_16S$Functional.group, betadisp_distances_16S$Functional.group.1, mat = TRUE, digits=3)
BD_16S.LS <- describe.by(betadisp_distances_16S$Life.span, betadisp_distances_16S$Life.span.1, mat = TRUE, digits=3)
#capture in single summary table
Betadispersion_16S_summary <- rbind(BD_16S.PlantID, BD_16S.Genus, BD_16S.Family, BD_16S.Order, BD_16S.Class, BD_16S.Provenance, BD_16S.PMA, BD_16S.NF, BD_16S.FG, BD_16S.LS)
```
#correlating 16S and ITS UniFrac distances with matK sequence distances
#merge samples and get mean ASV count
```{r}
physeq_ITS_merged <- merge_samples(physeq.rarefied_ITS, "PlantID", fun = mean)
physeq_16S_merged <- merge_samples(physeq.rarefied.16S, "PlantID", fun = mean)
```
##sample list in 16S and ITS ASV tables need to be in same order as pairwise distance matrix
#no matK data for C.secta or A.caesiigaluca so need to remove C.secta and A.caesiiglauca from ASV tables and reset order of plant species 
```{r import matk distances}
matK.dist <- read.csv("~/Xioben data/matK/matK_dist.csv")
row.names(matK.dist) = matK.dist$X
matK.dist = matK.dist[,-1]
```
#remove C.secta and A.caesiiglauca from phyloseq object
```{r, message=FALSE}
sample_data(physeq_ITS_merged)$PlantID <- sample_names(physeq_ITS_merged)
sample_data(physeq_16S_merged)$PlantID <- sample_names(physeq_16S_merged)
plants_to_remove <- c("A.caesiiglauca", "C.secta")
physeq_ITS_merged <- subset_samples(physeq_ITS_merged, !(PlantID %in% plants_to_remove))
physeq_16S_merged <- subset_samples(physeq_16S_merged, !(PlantID %in% plants_to_remove))
```
#calculate unifrac distances for new phyloseq objects
```{r}
Unifrac.its2 <- phyloseq::distance(physeq_ITS_merged, "wunifrac")
Unifrac.16s2 <- phyloseq::distance(physeq_16S_merged, "wunifrac")
```
# reorder matK table by plant species order in ASV table
```{r}
physeq_ITS_merged3 = as.data.frame(otu_table(physeq_ITS_merged))
physeq_16S_merged3 = as.data.frame(otu_table(physeq_16S_merged))
```
```{r}
matK.dist_ITS <- matK.dist[match(rownames(physeq_ITS_merged3), rownames(matK.dist)), ]
matK.dist_16S <- matK.dist[match(rownames(physeq_16S_merged3), rownames(matK.dist)), ]

matK.dist_ITS <- matK.dist_ITS[,order(colnames(matK.dist_ITS))]
matK.dist_ITS <- as.dist(as.matrix(matK.dist_ITS))
matK.dist_16S <- matK.dist_16S[,order(colnames(matK.dist_16S))]
matK.dist_16S <- as.dist(as.matrix(matK.dist_16S))
```
# matK mantel tests
```{r}
vegan::mantel(Unifrac.its2, matK.dist_ITS, method = "spearman", permutations = 999)
vegan::mantel(Unifrac.16s2, matK.dist_16S, method = "spearman", permutations = 999)
```
#hierarchial clustering of matK distances and unifrac distances
#16S data
```{r}
#16S community data
x16S.cluster <- hclust(Unifrac.16s2, method = "complete")
#MatK data
matK_16S.cluster <- hclust(matK.dist_16S, method = "complete")
```
#plotting 16S crt unifrac vs matK
```{r}
x16S.cluster2 <- ape::as.phylo(x16S.cluster)
matK_16S.cluster2 <- ape::as.phylo(matK_16S.cluster)
association_matrix <- cbind(x16S.cluster2$tip.label, matK_16S.cluster2$tip.label) #for connecting plant names
png(file="Fig_2.png", units="in", width=10, height=8, res=300)
cophyloplot(x16S.cluster2, matK_16S.cluster2, show.tip.label = TRUE, length.line = 6, space = 100, gap = 2, assoc = association, col = "red", lty = "dashed")
title(xlab="16S rRNA UniFrac distances vs Plant MatK distances")
dev.off()
```
#ITS data
```{r}
#ITS community data
ITS.cluster <- hclust(Unifrac.its2, method = "complete")
#MatK data
matK_ITS.cluster <- hclust(matK.dist_ITS, method = "complete")
```
#plotting ITS crt unifrac vs matK
```{r}
ITS.cluster2 <- ape::as.phylo(ITS.cluster)
matK_ITS.cluster2 <- ape::as.phylo(matK_ITS.cluster)
association <- cbind(ITS.cluster2$tip.label, matK_ITS.cluster2$tip.label)
png(file="Fig_1.png", units="in", width=10, height=8, res=300)
cophyloplot(ITS.cluster2, matK_ITS.cluster2, show.tip.label = TRUE, length.line = 6, space = 100, gap = 2, assoc = association, col = "red", lty = "dashed")
title(xlab="ITS UniFrac distances vs Plant MatK distances")
dev.off()
```
##Variance partitioning analysis
##ITS variance partitioning
```{r}
soil.env.ITS <- as.data.frame(sample.ITS2, row.names = rownames(sample.ITS2))
soil.env.ITS <- na.omit(soil.env.ITS)
plant.data.ITS <- data.frame(sample.its[,3:12])
rownames(plant.data.ITS) <- rownames(sample.its)
setdiff(rownames(plant.data.ITS), rownames(soil.env.ITS))
remove_samples2 <- c("kabir_ITS2rcbc368", "kabir_ITS2rcbc343", "kabir_ITS2rcbc354") #removing samples which aren't present in soil chemical data
plant.data.ITS <- plant.data.ITS[!(row.names(plant.data.ITS) %in% remove_samples2), ]
#remove unwanted samples from unifrac matrix
ITS.community <- as.matrix(unifrac.its)
setdiff(rownames(ITS.community), rownames(plant.data.ITS))
ITS.community <- ITS.community[!(row.names(ITS.community) %in% remove_samples2), ] #remove rows
ITS.community <- ITS.community[ , !(colnames(ITS.community) %in% remove_samples2)]
```
## Change the data frame with factors into numeric model matrix. Test effects of plant identity, plant life history, rhizosphere traits, soil chemical conditions, and stohcastic properties (residual variation)
```{r}
mm1_plantID <- model.matrix(~ PlantID, plant.data.ITS)[,-1] #plant identity
mm2_PLHT <- model.matrix(~ Provenance + Life.span + Functional.group, plant.data.ITS)[,-1] #plant life history traits
mm3_PRT <- model.matrix(~ Primary.mycorrhizal.association + Nitrogen.fixing, plant.data.ITS)[,-1] #plant rhizosphere traits
```
# Use RDA to test significance of each indivudal matrix whilst controlling for the other three matrices
```{r}
##testing full model
dbRDA.ALL <- vegan::capscale(ITS.community ~ mm1_plantID + mm2_PLHT + mm3_PRT + as.matrix(soil.env.ITS))
anova(dbRDA.ALL, step=200, perm.max=200) #fit full model

#Plant ID only
rda.result1 <- capscale(ITS.community ~ mm1_plantID + Condition(mm2_PLHT) + Condition(mm3_PRT) + Condition(as.matrix(soil.env.ITS)))
anova(rda.result1, step=200, perm.max=200) 

#soil parameters only
rda.result2 <- capscale(ITS.community ~ as.matrix(soil.env.ITS) + Condition(mm1_plantID) + Condition(mm2_PLHT) + Condition(mm3_PRT))
anova(rda.result2, step=200, perm.max=200)

#Life history parameters only. Non significant
rda.result3 <- capscale(ITS.community ~  mm2_PLHT + Condition(mm1_plantID) + Condition(mm3_PRT) + Condition(as.matrix(soil.env.ITS)))
anova(rda.result3, step=200, perm.max=200)

#Root associations only. Non significant
rda.result4 <- capscale(ITS.community ~  mm3_PRT + Condition(mm1_plantID) + Condition(mm2_PLHT) + Condition(as.matrix(soil.env.ITS)))
anova(rda.result4, step=200, perm.max=200)  
```
##varpart model for ITS data
```{r}
its.varpart <- varpart(ITS.community, mm1_plantID, mm2_PLHT, mm3_PRT, soil.env.ITS)
tiff("ITS_VarPart.tiff", units="in", width=8, height=5, res=300)
plot(its.varpart, digits = 1, Xnames = c('PI***','PLH', 'RS', 'SC***'), bg = c('navy', 'purple', 'red', 'yellow'), cex=1.2, id.size=1.2)
dev.off()
```
##16S variance partitioning
```{r}
soil.env.16S <- as.data.frame(sample.16S2, row.names = rownames(sample.16S))
soil.env.16S <- na.omit(soil.env.16S)
plant.data.16S <- data.frame(sample.16S[,3:12])
rownames(plant.data.16S) <- rownames(sample.16S)
setdiff(rownames(plant.data.16S), rownames(soil.env.16S))
remove_samples2 <- c("X614", "X621", "X786") #remvoing samples which aren't present in soil chem data
plant.data.16S <- plant.data.16S[!(row.names(plant.data.16S) %in% remove_samples2), ]

#remove unwanted samples from unifrac matrix
x16S.community <- as.matrix(unifrac.16S)
setdiff(rownames(x16S.community), rownames(plant.data.16S))
x16S.community <- x16S.community[!(row.names(x16S.community) %in% remove_samples2), ] #remove rows
x16S.community <- x16S.community[ , !(colnames(x16S.community) %in% remove_samples2)]
```
##Change the data frame with factors into numeric model matrix. Test effects of plant identity, plant life history, rhizosphere traits, soil chemical conditions, and stohcastic properties (residual variation)
```{r}
mm1_16S_plantID <- model.matrix(~ PlantID, plant.data.16S)[,-1] #plant identity
mm2_16S_PLHT <- model.matrix(~ Provenance + Life.span + Functional.group, plant.data.16S)[,-1] #plant life history traits
mm3_16S_PRT <- model.matrix(~ Primary.mycorrhizal.association + Nitrogen.fixing, plant.data.16S)[,-1] #plant rhizosphere traits
```
# Use RDA to test sigificanance of each indivudal matrix whilst controlling for the other three matrices
```{r}
##testing full model
dbRDA.16S.ALL <- capscale(x16S.community ~ mm1_16S_plantID + mm2_16S_PLHT + mm3_16S_PRT + as.matrix(soil.env.16S))
anova(dbRDA.16S.ALL, step=200, perm.max=200) #fit full model

#Plant ID only. Conditioning for life history, plant root behaviour and soil env properties
rda.16S.result1 <- capscale(x16S.community ~ mm1_16S_plantID + Condition(mm2_16S_PLHT) + Condition(mm3_16S_PRT) + Condition(as.matrix(soil.env.16S)))
anova(rda.16S.result1, step=200, perm.max=200) 

#soil parameters only. Conditioning for plantID, plant root behavior and life history
rda.16S.result2 <- capscale(x16S.community ~ as.matrix(soil.env.16S) + Condition(mm1_16S_plantID) + Condition(mm2_16S_PLHT) + Condition(mm3_16S_PRT))
anova(rda.16S.result2, step=200, perm.max=200)

#Life history parameters only. Non significant. Remove from final varpart model
rda.16S.result3 <- capscale(x16S.community ~  mm2_16S_PLHT + Condition(mm1_16S_plantID) + Condition(mm3_16S_PRT) + Condition(as.matrix(soil.env.16S)))
anova(rda.16S.result3, step=200, perm.max=200)

#Root associations only. Non significant. Remove from final varpart model
rda.16S.result4 <- capscale(x16S.community ~  mm3_16S_PRT + Condition(mm1_16S_plantID) + Condition(mm2_16S_PLHT) + Condition(as.matrix(soil.env.16S)))
anova(rda.16S.result4, step=200, perm.max=200) 
```
##final 16S varpart model 
```{r}
x16s.varpart <- varpart(x16S.community, mm1_16S_plantID, mm2_16S_PLHT, mm3_16S_PRT, soil.env.16S)
tiff("16S_VarPart.tiff", units="in", width=8, height=5, res=300)
plot(x16s.varpart, digits = 1, Xnames = c('PI***','PLH', 'RS', 'SC***'), bg = c('navy', 'purple', 'red', 'yellow'), cex=1.2, id.size=1.2)
dev.off()
```
#Forward selection of soil chemical parameters using ordiR2step
##16S
```{r}
rda.16S.0 <- capscale(x16S.community ~ 1, data = soil.env.16S)
rda.16S.all <- capscale(x16S.community ~ pH + Olsen.Phosphorus + Sulphate.Sulphur + Pot..Avail..N + Anaer..Min..N +        Anaer..Min..N.Tot..N + Organic.Matter + Total.Carbon + Total.Nitrogen + C.N.Ratio + Volume.Weight, data = soil.env.16S)
adjR2.all <- RsquareAdj(rda.16S.all)$adj.r.squared
rda.16S.sel <- ordiR2step(rda.16S.0, scope = formula(rda.16S.all), R2scope = adjR2.all, direction= "forward", permutations = 999)
rda.16S.selR2_adj <- rda.16S.sel
rda.16S.selR2_adj$anova$`Pr(>F)` <- p.adjust(rda.16S.selR2_adj$anova$`Pr(>F)`, method = 'bonferroni', n = ncol(soil.env.16S))
x16S_ForwardSel <- rda.16S.selR2_adj$anova
```
##16S capscale using plant species olsen P, pH and sulphate sulphute
```{r}
dbRDA.16S <- capscale(x16S.community ~ Olsen.Phosphorus + pH + Sulphate.Sulphur, data = soil.env.16S)
anova(dbRDA.16S) #just FS soil chem

dbRDA.16S1 <- capscale(x16S.community ~ Olsen.Phosphorus + pH + Sulphate.Sulphur, mm1_16S_plantID, data = soil.env.16S)
anova(dbRDA.16S1) #FS soil chem + plant ID
```
##ITS
```{r}
rda.ITS.0 <- capscale(ITS.community ~ 1, data = soil.env.ITS)
rda.ITS.all <- capscale(ITS.community ~ pH + Olsen.Phosphorus + Sulphate.Sulphur + Pot..Avail..N + Anaer..Min..N +        Anaer..Min..N.Tot..N + Organic.Matter + Total.Carbon + Total.Nitrogen + C.N.Ratio + Volume.Weight, data = soil.env.ITS)
adjR2.all <- RsquareAdj(rda.ITS.all)$adj.r.squared
rda.ITS.sel <- ordiR2step(rda.ITS.0, scope = formula(rda.ITS.all), R2scope = adjR2.all, direction= "forward", permutations = 999)
rda.ITS.selR2_adj <- rda.ITS.sel
rda.ITS.selR2_adj$anova$`Pr(>F)` <- p.adjust(rda.ITS.selR2_adj$anova$`Pr(>F)`, method = 'bonferroni', n = ncol(soil.env.ITS))
ITS_ForwardSel <- rda.ITS.selR2_adj$anova
```
##ITS capscale using plant species olsen P, pH and sulphate sulphute
```{r}
dbRDA.ITS <- vegan::capscale(ITS.community ~ Olsen.Phosphorus + Anaer..Min..N.Tot..N + Volume.Weight, data = soil.env.ITS)
anova(dbRDA.ITS)

dbRDA.ITS1 <- vegan::capscale(ITS.community ~ Olsen.Phosphorus + Anaer..Min..N.Tot..N + Volume.Weight + mm1_plantID, data = soil.env.ITS)
anova(dbRDA.ITS1)
```
#PIME: Prevalance Interval for Microbiome Evalution https://bocasbiome.github.io/wf7.html; https://github.com/microEcology/pime; https://doi.org/10.1111/1755-0998.13116
```{r}
#remotes::install_github("microEcology/pime")
library(pime)
```
#Remove phytree from phyloseq object or else analysis won't run
```{r create 16S phyloseq object}
physeq.rarefied_ITS@phy_tree <- NULL
physeq.rarefied.16S@phy_tree <- NULL
```
#Prediction using random forests on full dataset. Results in Out of Bag error rate. Baseline noise detection. 
```{r}
pime.oob.error(physeq.rarefied_ITS, "PlantID") ##OOB error rate 0.7889447
pime.oob.error(physeq.rarefied.16S, "PlantID") ##OOB error rate 0.8636364
```
#split dataset by variable.
```{r}
per_variable_ITS = pime.split.by.variable(physeq.rarefied_ITS, "PlantID")
per_variable_16S = pime.split.by.variable(physeq.rarefied.16S, "PlantID")
```
#calculate highest possible prevalance intervals
```{r}
prevalences_ITS = pime.prevalence(per_variable_ITS)
head(prevalences_ITS)
prevalences_16S = pime.prevalence(per_variable_16S)
head(prevalences_16S)
```
#calculate best prevalence interval for the datset
```{r}
#its
set.seed(42)
best.prev.ITS = pime.best.prevalence(prevalences_ITS, "PlantID") #0% OOB from prevalence 80% with 188 ASVs
best.prev.ITS$`OOB error`

#16s
set.seed(42)
best.prev.16S = pime.best.prevalence(prevalences_16S, "PlantID") #0% OOB from prevalence 90% with 544 ASVs
best.prev.16S$`OOB error`
```
#return results from random forest classification. table with OTU/ASV importance of the chosen prevalence interval. Pime keeps only the top 30 OTUs/ASVs, with highest MDA.
```{r}
imp75_ITS = best.prev.ITS$`Importance`$`Prevalence 75`
imp80_16S = best.prev.16S$`Importance`$`Prevalence 80`
```
#using prevalence 80 and 90
#obtain phyloseq object at desired cutoff
```{r}
prevalence.75_ITS = prevalences_ITS$`75`
prevalence.80_16S = prevalences_16S$`80`
```
#estimating prediction error
```{r}
randomized_ITS = pime.error.prediction(physeq.rarefied_ITS, "PlantID", bootstrap = 100, parallel = TRUE, max.prev = 95)
randomized_ITS$Plot
#estimate OOB error with each prevalence interval filtering
replicated.oob.error_ITS = pime.oob.replicate(prevalences_ITS, "PlantID", bootstrap = 100, parallel = TRUE)
randomized_16S = pime.error.prediction(physeq.rarefied.16S, "PlantID", bootstrap = 100, parallel = TRUE, max.prev = 95)
randomized_16S$Plot
#estimate OOB error with each prevalence interval filtering
replicated.oob.error_16S = pime.oob.replicate(prevalences_16S, "PlantID", bootstrap = 100, parallel = TRUE)
```
##idenitfy ITS ASVs which significantly vary across plant species
```{r}
#prevalence.75_ITS_genus <- tax_glom(prevalence.75_ITS, taxrank = "Genus", NArm = FALSE)
pime_metaSeq_ITS <- phyloseq_to_metagenomeSeq(prevalence.75_ITS)
##Computes a log-normal linear model and permutation based p-values
plantID = pData(pime_metaSeq_ITS)$PlantID
mod = model.matrix(~ 0 + plantID)
fit_ITS = fitLogNormal(obj = pime_metaSeq_ITS, useCSSoffset = FALSE, mod=mod, B=999) #fit model
adjustedPvalues = p.adjust(fit_ITS$p, method = "holm") #adj p values for multiple comparisons
##extract results
Pime_lmFit_t <- as.data.frame(fit_ITS$t) #model t values
Pime_lmFit_p.adj <- as.data.frame(adjustedPvalues) #adjusted p values
Pime_lmFit_FC <- as.data.frame(fit_ITS$fit$coefficients)
Pime_lmFit_FC <- cbind(Pime_lmFit_FC, Pime_lmFit_p.adj)
#get taxa names for reference
lmFit_ITS_taxa <- as.data.frame(tax_table(prevalence.75_ITS))
lmFit_ITS_taxa['ASV_ID'] <- rownames(lmFit_ITS_taxa)
#match order of ASVs
ordered_ITS_taxa <- rownames(Pime_lmFit_t)
lmFit_ITS_taxa2 <- lmFit_ITS_taxa[match(ordered_ITS_taxa, lmFit_ITS_taxa$ASV_ID),]
```
#filter ITS ASVs from phyloseq object which do not sig vary by plant ID
```{r}
Pime_lmFit_p.adj["ASV"] <- rownames(Pime_lmFit_p.adj)
Pime_lmFit_p.adj <- Pime_lmFit_p.adj[Pime_lmFit_p.adj$adjustedPvalues <= "0.05", ] #select significant p-values
keep_its_asvs <- Pime_lmFit_p.adj$ASV
prevalence.75_ITS3 <- prune_taxa(keep_its_asvs, prevalence.75_ITS)
ITS_sig_taxa <- tax_table(prevalence.75_ITS3) #taxanames to be plotted later on in the heatmap
```
#16s DA analysis
```{r}
pime_metaSeq_16S2 <- phyloseq_to_metagenomeSeq(prevalence.80_16S)
##Computes a log-normal linear model and permutation based p-values
plantID2 = pData(pime_metaSeq_16S)$PlantID
mod2 = model.matrix(~ 0 + plantID2)
fit_16S = fitLogNormal(obj = pime_metaSeq_16S, useCSSoffset = FALSE, mod=mod2, B=999) #fit model
adjustedPvalues2 = p.adjust(fit_16S$p, method = "holm") #adj p values for multiple comparisons
##extract results
Pime_lmFit_t2 <- as.data.frame(fit_16S$t) #model t values
Pime_lmFit_p.adj2 <- as.data.frame(adjustedPvalues2) #adjusted p values
Pime_lmFit_FC2 <- as.data.frame(fit_16S$fit$coefficients)
Pime_lmFit_FC2 <- cbind(Pime_lmFit_FC2, Pime_lmFit_p.adj2)
```
#filter 16S ASVs from phyloseq object which do not sig vary by plant ID
```{r}
Pime_lmFit_p.adj2["ASV"] <- rownames(Pime_lmFit_p.adj2)
Pime_lmFit_p.adj2 <- Pime_lmFit_p.adj2[Pime_lmFit_p.adj2$adjustedPvalues2 <= "0.05", ] #select significant p-values
keep_16s_asvs <- Pime_lmFit_p.adj2$ASV
prevalence.80_16S3 <- prune_taxa(keep_16s_asvs, prevalence.80_16S) 
x16S_sig_taxa <- tax_table(prevalence.80_16S3) #taxanames to be plotted later on in the heatmap
```
##plotting taxonomic differences 
```{r}
#install.packages("pheatmap")
library(pheatmap)
```
##ITS
```{r}
#remove non sig taxa 
Pime_lmFit_FC <- dplyr::filter(Pime_lmFit_FC, adjustedPvalues < 0.05)
Pime_lmFit_FC <- Pime_lmFit_FC[,-38]
colnames(Pime_lmFit_FC) <- c("A.caesiiglauca" , "A.capillaris" , "A.dealbata" , "A.glutinosa", "A.inermis", "A.lessoniana", "A.millefolium", "B.greyi", "C.conspicua", "C.robusta", "C.secta", "C.vulgare", "D.glomerata", "E.vulgare", "F.novae-zelandiae ", "H.lanatus", "H.odora", "H.perforatum", "L.aboreus", "L.perenne", "M.astonii", "M.complexa", "M.sativa", "O.leptophyllus", "O.virgata", "P.cita", "P.colensoi", "P.contorta", "P.cookianum", "P.radiata", "P.totara", "R.acetosella", "R.obtusifolius", "S.microphylla", "T.pratense", "T.repens", "U.europaeus")

rownames(Pime_lmFit_FC) <- c("s_Mortierella_pseudozygospora [ASV_31]","g_Mortierella [ASV_10]","g_Mortierella [ASV_16]","s_Mortierella_gamsii [ASV_7]","s_Mortierella_gamsii [ASV_8]","s_Mortierella_sarnyensis [ASV_29]","s_Mortierella_humilis [ASV_3]","g_Mortierella [ASV_4]","s_Apiotrichum_porosum [ASV_1]","g_Apiotrichum [ASV_34]","s_Solicoccozyma_terrea [ASV_13]","g_Solicoccozyma [ASV_54]","s_Solicoccozyma_terricola [ASV_5]","o_Leucosporidiales [ASV_59]","s_Cladosporium_delicatulum [ASV_48]","s_Exophiala_pisciphila [ASV_53]","s_Saitozyma_podzolica [ASV_6]","s_Pseudogymnoascus_destructans [ASV_2]","s_Geomyces_asperulatus [ASV_22]","s_Ilyonectria_mors-pacis [ASV_49]","s_Fusarium_acutatum [ASV_17]","s_Umbelopsis_dimorpha [ASV_606]","s_Mortierella_gamsii [ASV_12]","g_Mortierella [ASV_57]","s_Mortierella_echinula [ASV_260]","s_Mortierella_globulifera [ASV_47]","g_Apiotrichum [ASV_90]","g_Solicoccozyma [ASV_37]","s_Exophiala_equi [ASV_52]","s_Saitozyma_podzolica [ASV_46]","c_Leotiomycetes [ASV_142]","o_Helotiales [ASV_426]","s_Russula_delica [ASV_229]","f_Stachybotryaceae [ASV_198]","s_Acremonium_curvulum [ASV_224]","p_Ascomycota [ASV_169]")
Pime_lmFit_FC.t = t(Pime_lmFit_FC) #transpose for heatmap
#heatmap
tiff("ITS_heatmap.tiff", units="in", width=12, height=10, res=300)
ITS_heatmap <- pheatmap(Pime_lmFit_FC.t, cex=0.9) #plant species and asvs clustered via hclust
dev.off()
```
##plotting 16S taxonomic heatmap
```{r}
#remove non sig taxa 
#edit col names
Pime_lmFit_FC2 <- dplyr::filter(Pime_lmFit_FC2, adjustedPvalues2 < 0.05)
Pime_lmFit_FC2 <- Pime_lmFit_FC2[,-38]
colnames(Pime_lmFit_FC2) <- c("A.caesiiglauca" , "A.capillaris" , "A.dealbata" , "A.glutinosa", "A.inermis", "A.lessoniana", "A.millefolium", "B.greyi", "C.conspicua", "C.robusta", "C.secta", "C.vulgare", "D.glomerata", "E.vulgare", "F.novae-zelandiae ", "H.lanatus", "H.odora", "H.perforatum", "L.aboreus", "L.perenne", "M.astonii", "M.complexa", "M.sativa", "O.leptophyllus", "O.virgata", "P.cita", "P.colensoi", "P.contorta", "P.cookianum", "P.radiata", "P.totara", "R.acetosella", "R.obtusifolius", "S.microphylla", "T.pratense", "T.repens", "U.europaeus")
#edit rownames
rownames(Pime_lmFit_FC2) <- c("g_Ktedonobacter [ASV_11]", "c_Actinobacteria [ASV_9]","g_Conexibacter [ASV_31]",
"g_Actinoallomurus [ASV_13]","g_Jatrophihabitans [ASV_100]","g_Mycobacterium [ASV_49]","f_Chitinophagaceae [ASV_5]","p_candidate_division_WPS-1 [ASV_57]","c_Spartobacteria [ASV_3]","c_Acidobacteria_Gp2 [ASV_60]","c_Acidobacteria_Gp1 [ASV_29]","g_Bacillus [ASV_15]","g_Devosia [ASV_6]","f_Caulobacteraceae [ASV_82]","g_Roseiarcus [ASV_7]","o_Rhizobiales [ASV_12]","g_Pseudolabrys [ASV_52]","g_Bradyrhizobium [ASV_1]","f_Rhodospirillaceae [ASV_44]","g_Porphyrobacter [ASV_21]","g_Rhodanobacter [ASV_17]","g_Povalibacter [ASV_22]","c_Acidobacteria_Gp16 [ASV_67]","g_Bellilinea [ASV_2427]","g_Gaiella [ASV_45]","f_Iamiaceae [ASV_261]","g_Kribbella [ASV_464]","f_Micromonosporaceae [ASV_193]","g_Arthrobacter [ASV_8]","g_Cellulomos [ASV_639]","g_Phaeodactylibacter [ASV_1605]","g_Flavitalea [ASV_18]","g_Terrimos [ASV_479]","g_Flavobacterium [ASV_292]","c_Acidobacteria_Gp6 [ASV_131]","c_Acidobacteria_Gp4 [ASV_162]","o_Aridibacter [ASV_441]","c_Acidobacteria_Gp4 [ASV_366]","c_Acidobacteria_Gp7 [ASV_145]","g_Aquisphaera [ASV_112]","f_Verrucomicrobiaceae [ASV_2407]","o_Terrimicrobium [ASV_211]","c_Subdivision3 [ASV_24]","o_Edaphobacter [ASV_101]","c_Acidobacteria_Gp5 [ASV_741]","o_Telmatobacter [ASV_32]","o_Candidatus_Solibacter [ASV_87]","f_Polyangiaceae [ASV_116]","g_Minicystis [ASV_252]","f_Nannocystaceae [ASV_303]","o_Myxococcales [ASV_19]","g_Sporosarci [ASV_43]","g_Aminobacter [ASV_206]","g_Rhodomicrobium [ASV_154]","f_Acetobacteraceae [ASV_160]","g_Acidisoma [ASV_71]","f_Reyranella [ASV_55]","c_Alphaproteobacteria [ASV_39]","g_Sphingomos [ASV_16]","g_Formivibrio [ASV_51]","o_Burkholderiales [ASV_88]","g_Variovorax [ASV_183]","c_Betaproteobacteria [ASV_58]","g_Dokdonella [ASV_78]","g_Thermomos [ASV_167]","o_Xanthomonadales [ASV_118]","c_Gammaproteobacteria [ASV_108]","f_Thermomonosporaceae [ASV_109]","g_Pirellula [ASV_260]","o_Acidobacterium [ASV_207]","c_Acidobacteria_Gp3 [ASV_286]","g_Phenylobacterium [ASV_239]","f_Rhizomicrobium [ASV_27]","g_Gemmatimos [ASV_83]","o_Acidimicrobiales [ASV_128]","o_Granulicella [ASV_318]","g_Rhodoplanes [ASV_56]","g_Arenimos [ASV_143]")
Pime_lmFit_FC2.t = t(Pime_lmFit_FC2) #transpose for heatmap
#heatmap
tiff("16S_heatmap.tiff", units="in", width=15, height=12, res=300)
x16S_heatmap <- pheatmap(Pime_lmFit_FC2.t, cex=0.9) #plant species and asvs clustered via hclust
dev.off()
```
###plotting heatmaps correlating plants on 16S and ITS asv log fold change
##make heatmap/perform analysis for each metadata factor.
```{r}
Pime_lmFit_FoldChange <- as.data.frame(fit_ITS$fit$coefficients)
Pime_lmFit_FoldChange <- Pime_lmFit_FoldChange[,-38]
Pime_lmFit_FoldChange2 <- as.data.frame(fit_16S$fit$coefficients)
Pime_lmFit_FoldChange2 <- Pime_lmFit_FoldChange2[,-38]
colnames(Pime_lmFit_FoldChange) <- c("A.caesiiglauca" , "A.capillaris" , "A.dealbata" , "A.glutinosa", "A.inermis", "A.lessoniana", "A.millefolium", "B.greyi", "C.conspicua", "C.robusta", "C.secta", "C.vulgare", "D.glomerata", "E.vulgare", "F.novae-zelandiae ", "H.lanatus", "H.odora", "H.perforatum", "L.aboreus", "L.perenne", "M.astonii", "M.complexa", "M.sativa", "O.leptophyllus", "O.virgata", "P.cita", "P.colensoi", "P.contorta", "P.cookianum", "P.radiata", "P.totara", "R.acetosella", "R.obtusifolius", "S.microphylla", "T.pratense", "T.repens", "U.europaeus")
colnames(Pime_lmFit_FoldChange2) <- c("A.caesiiglauca" , "A.capillaris" , "A.dealbata" , "A.glutinosa", "A.inermis", "A.lessoniana", "A.millefolium", "B.greyi", "C.conspicua", "C.robusta", "C.secta", "C.vulgare", "D.glomerata", "E.vulgare", "F.novae-zelandiae ", "H.lanatus", "H.odora", "H.perforatum", "L.aboreus", "L.perenne", "M.astonii", "M.complexa", "M.sativa", "O.leptophyllus", "O.virgata", "P.cita", "P.colensoi", "P.contorta", "P.cookianum", "P.radiata", "P.totara", "R.acetosella", "R.obtusifolius", "S.microphylla", "T.pratense", "T.repens", "U.europaeus")
```
#correlation plot between plant species 
#format dataframes for plotting
```{r}
#library(reshape2)
plantID_cormat <- cor(Pime_lmFit_FoldChange)
head(plantID_cormat)

  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cormat){
    cormat[lower.tri(cormat)]<- NA
    return(cormat)
  }
#formatting fungi 
upper_tri <- get_upper_tri(plantID_cormat)
plantID_cormat.m <- melt(upper_tri, na.rm = TRUE)
head(plantID_cormat.m)
#formatting bacteria
plantID_cormat2 <- cor(Pime_lmFit_FoldChange2)
upper_tri2 <- get_upper_tri(plantID_cormat2)
plantID_cormat.m2 <- melt(upper_tri2, na.rm = TRUE)
head(plantID_cormat.m2)
```
```{r}
##fungi corrplot
PlantID_corplot <- ggplot(data = plantID_cormat.m, aes(x=Var1, y=Var2, fill=value)) + 
  geom_tile(color = "white")+
  xlab("Plant species")+
  ylab("Plant species")+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 0, limit = c(-1,1), space = "Lab", 
   name="Pearson\nCorrelation") +
  theme_minimal()+ 
 theme(axis.text.x = element_text(angle = 90, vjust = 1, 
    size = 8, hjust = 1), axis.text.y = element_text(angle = 0, vjust = 1, 
    size = 8, hjust = 1))+
 coord_fixed()
tiff("~/Xioben data/Alexa_TRM_analysis/April_2024/Fig_6.tiff", units="in", width=10, height=10, res=300)
PlantID_corplot
```
##extracting p-values for correlation matrices
```{r}
flat_cor_mat <- function(cor_r, cor_p){
  library(tidyr)
  library(tibble)
  cor_r <- rownames_to_column(as.data.frame(cor_r), var = "row")
  cor_r <- gather(cor_r, column, cor, -1)
  cor_p <- rownames_to_column(as.data.frame(cor_p), var = "row")
  cor_p <- gather(cor_p, column, p, -1)
  cor_p_matrix <- left_join(cor_r, cor_p, by = c("row", "column"))
  cor_p_matrix
}
```
##extracting correlation data
```{r}
##fungi
SpeciesCor <- Hmisc::rcorr(as.matrix(Pime_lmFit_FoldChange))
Species_cor_matrix <- flat_cor_mat(SpeciesCor$r, SpeciesCor$P)
##bacteria
SpeciesCor2 <- Hmisc::rcorr(as.matrix(Pime_lmFit_FoldChange2))
Species_cor_matrix2 <- flat_cor_mat(SpeciesCor2$r, SpeciesCor2$P)
#getting average and SD R value
fungi_R <- Species_cor_matrix$cor
fungi_R <- fungi_R[fungi_R < 1] #removing values of 1 (self correlation between same plant species)
mean(fungi_R) #mean 0.686
sd(fungi_R) #SD 0.078
bacteria_R <- Species_cor_matrix2$cor
bacteria_R <- bacteria_R[bacteria_R < 1] #removing values of 1 (self correlation between same plant species)
mean(bacteria_R) #mean 0.698
sd(bacteria_R) #SD 0.082
```
